source/app/libpd.c

```diff
@@ -135,9 +139,15 @@ static void parodus_receive()
                                 res_wrp_msg->u.req.payload_size = strlen(res_wrp_msg->u.req.payload);
```

Comment:
Move WalPrint("Response payload is %s\n",(char *)(res_wrp_msg->u.req.payload)); inside NULL check

Reply:
Incorporated this change and amended the review.